provider-integration/integration-module/src/main/kotlin/plugins/compute/ucloud/FeatureFileMount.kt

```diff
@@ -144,7 +158,15 @@ class FeatureFileMount(
                 ) {
                     joinPath(pathSplit[pathSplit.size - 2], pathSplit.last())
                 } else {
-                    pathSplit.last()
+                    var path = mount.fileName
+                    val driveId = mount.path.normalize().components().getOrNull(1) ?: error("Unexpected path: $path")
+                    if(pathSplit.size == 2 && path == driveId) {
+                        val driveName = runBlocking {
+                            collectionCache.get(mount.path.normalize().components().getOrNull(1)!!)?.specification?.title ?: error("Unexpected Drive")
+                        }
```

Comment:
I don't understand why this is `runBlocking`. The function itself is allowed to suspend. We should re-use the `driveId` instead of recomputing it again.

```suggestion
                        val driveName = collectionCache.get(driveId)?.specification?.title ?: error("Unexpected Drive")
```